pulser/simulation/simconfig.py

```diff
@@ -51,55 +54,59 @@ def __str__(self) -> str:
         s = ""
         s += "Options:\n"
         s += "----------       \n"
-        s += "noise types:       " + str(self.noise) + "\n"
-        s += "spam dictionary:   " + str(self.spam_dict) + "\n"
-        s += "temperature:       " + str(self.temperature) + "K" + "\n"
-        s += "number of runs:    " + str(self.runs) + "\n"
-        s += "samples per runs:  " + str(self.samples_per_run) + "\n"
-        s += "laser waist:       " + str(self.laser_waist) + "μm \n"
+        s += "Noise types:       " + str(self.noise) + "\n"
+        s += "Spam dictionary:   " + str(self.spam_dict) + "\n"
+        s += "Temperature:       " + str(self.temperature) + "K" + "\n"
+        s += "Number of runs:    " + str(self.runs) + "\n"
+        s += "Samples per runs:  " + str(self.samples_per_run) + "\n"
+        s += "Laser waist:       " + str(self.laser_waist) + "μm \n"
         s += "\n" + "Solver Options: \n"+str(self.solver_options)[10:-1]+"\n"
         return s
 
     @property
-    def noise(self) -> List[str]:
+    def noise(self) -> list[NOISE_TYPES]:
         return self._noise
 
     @noise.setter
-    def noise(self, noise_types: List[str]) -> None:
-        self._noise: List[str] = []
+    def noise(self, noise_types: abc.Sequence[NOISE_TYPES]) -> None:
+        self._noise: list[NOISE_TYPES] = []
         for noise_type in noise_types:
             self._add_noise(noise_type)
 
-    def _add_noise(self, noise_type: str) -> None:
-        """Adds a noise model to the SimConfig instance, to be used
-            in Simulation.
+    def _add_noise(self, noise_type: NOISE_TYPES) -> None:
+        """Adds a noise model to the SimConfig instance.
+
             Args:
-                noise_type (str): Choose among:
-                    'dephasing': random phase (Z) flip
-                    'doppler': Noisy doppler runs
-                    'amplitude': Noisy gaussian beam
-                    'SPAM': SPAM errors. Adds:
-                        eta: Probability of each atom to be badly prepared
-                        epsilon: false positives
-                        epsilon_prime: false negatives
+                noise_type (NOISE_TYPES): Choose among:
+                -   'dephasing': Random phase (Z) flip
+                -   'doppler': Noisy doppler runs
+                -   'amplitude': Noisy gaussian beam
+                -   'SPAM': SPAM errors. Adds:
+                    --  eta: Probability of each atom to be badly prepared
+                    --  epsilon: Probability of false positives
+                    --  epsilon_prime: Probability of false negatives.
```

Comment:
It would also be good to given the user access to this information. In a private method, they are hidden away.